### PR TITLE
CSM Replication Support For PowerScale

### PR DESCRIPTION
**Pull request by IliaRN**
# PR Submission checklist

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/116|

# Common PR Checklist:

- [x] Have you made sure that the code compiles?
- [x] Have you commented your code, particularly in hard-to-understand areas
- [ ] Did you run tests in a real Kubernetes cluster?
- [x] Have you maintained backward compatibility

## Description of your changes:
This PR addresses an issue with CSM Replication Support For PowerScale. Added initial replication code and DeletePolicy method.

### CODE REVIEW

**Review by walker2 — APPROVED** [review]

File: api/v11/api_v11_replication.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+// GetPolicyByName returns policy by name
+func GetPolicyByName(
+	ctx context.Context,
+	client api.Client, name string) (policy *Policy, err error) {
+	p := &Policies{}
+	err = client.Get(ctx, policiesPath, name, nil, nil, &p)
+	if err != nil || len(p.Policy) == 0 {
+		return nil, err
```
Comment (randeepsharma) [inline api/v11/api_v11_replication.go:36]: In a scenario where err is nil and len(p.Policy)==0 , we will be sending nil, nil essentially.
May be we can consider creating an err and returning it.

**Comment by randeepsharma** [comment]
Need to rebase again, this PR shows changes that are already merged.

**Review by walker2 — APPROVED** [review]

**Review by bpjain2004 — APPROVED** [review]

**Comment by prablr79** [comment]
@IliaRN can you check few comments ? post addressing that we can merge this PR.

**Review by randeepsharma — APPROVED** [review]
LGTM

**Comment by prablr79** [comment]
@randeepsharma is rebase is done for this PR ?

**Comment by randeepsharma** [comment]
> @randeepsharma is rebase is done for this PR ?

Yes

**Pull request merged**